src/lib.rs

```diff
@@ -210,7 +210,10 @@ fn grid_points(bounds: &Bounds, grid_size: usize) -> HashMap<(i8, i8), (usize, u
     let mut points = HashMap::new();
     for x in 1..grid_size {
         for y in 1..grid_size {
-            points.insert((x as i8, y as i8), (x * x_width, y * y_width));
+            points.insert(
+                (x as i8, y as i8),
+                (x * x_width + bounds.lower_x, y * y_width + bounds.lower_y)
```

Comment:
While this does fix a bug, I'm not yet convinced that it fixes #1. `square_edge` is always at least 1, and `pixel_average` will move a further 1 from the point, so for very thin images or very high `grid_size`s I think we can still run into a case where we insert a point here with `x * x_width + bounds.lower_x = 1` (or similar for `y`) and then subtract at least 2.

Reply:
Added `clamp`s below to always keep us in bounds. With this change I can process 1x1 pixel images without errors.